lib/index.js

```diff
@@ -0,0 +1,89 @@
+/*!
+ * Copyright (c) 2017 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+const bedrock = require('bedrock');
+const brLedger = require('bedrock-ledger');
+const equihashSigs = require('equihash-signature');
+const validate = require('bedrock-validation').validate;
+const BedrockError = bedrock.util.BedrockError;
+
+require('./config');
+
+bedrock.events.on('bedrock.start', () => brLedger.use('EquihashValidator2017', {
+  type: 'validator',
+  api: api
+}));
+
+const api = {};
+// NOTE: only exported for tests
+module.exports = api;
+
+api.mustValidateEvent = (event, validatorConfig, options, callback) => {
+  if(typeof options === 'function') {
+    callback = options;
+    options = {};
+  }
+  api.validateConfiguration(validatorConfig, err => {
+    if(err) {
+      return callback(err);
+    }
+    if(!validatorConfig.eventFilter.some(f =>
+      f.type === 'EventTypeFilter' && f.eventType.includes(event.type))) {
+      return callback(null, false);
+    }
+    callback(null, true);
+  });
+};
+
+api.validateConfiguration = (validatorConfig, callback) =>
+  validate('ledger-validator-equihash-config', validatorConfig, callback);
+
+api.validateEvent = (event, validatorConfig, callback) => {
+  if(!validatorConfig.eventFilter.some(f =>
```

Comment:
I don't think this is necessary. The contract that a consensus method has with this validator is that it will always call `mustValidateEvent` prior to calling `validateEvent`, so any checks performed there should not need to be repeated here -- and we should be guaranteed, based on the implementation above, to have passed any event filter.